bin/pycbc_live

```diff
@@ -403,6 +401,32 @@ class LiveEventManager(object):
             opt_scheme = args.processing_scheme.split(':')[0]
             cmd += '--processing-scheme {}:1 '.format(opt_scheme)
 
+        # Save the command which would be used:
+        snroc_fname = os.path.join(out_dir_path, 'snr_optimize_command.txt')
+        with open(snroc_fname,'w') as snroc_file:
+            snroc_file.write(cmd)
+
+        return cmd, out_dir_path
+
+    def run_optimize_snr(self, cmd, out_dir_path, attribute_fname, gid):
+        """
+        Run the optimize_snr instance setup up by setup_optimize_snr
+
+        Parameters
+        ----------
+        cmd: str
+            Command to tell subprocess what to run for the
+            optimize_snr process
+        out_dir_path: os.path or str
+            Place for storing the SNR optimization results and log
+        attribute_fname: str
+            The filename of the attributes of the candidate
+        gid: str
+            GraceDB ID of the candidate
+        """
+        with h5py.File(attribute_fname, 'a') as hdfp:
+            if gid is not None:
+                hdfp['gid'] = gid
```

Comment:
The `with` and `if` blocks should be swapped. We do not want to open the file unless we are sure we are going to use it.